tests/test_api_types.py

```diff
@@ -1,5 +1,15 @@
+from typing import TYPE_CHECKING
+
 import pytest
-from pydantic import ValidationError
+
+if TYPE_CHECKING:  # mypy really does not like this conditional import.
+    import pydantic
```

Comment:
Should this be `import pydantic.v1 as pydantic`, since mypy is going to run in a venv with `pydantic>=2`?

I'm not sure it winds up making a difference because we now have `# type: ignore` everywhere, but it seems "more correct", and perhaps it will make a difference at some point in the future.

Reply:
This would cause a mypy error for v1 users, would it not?

Reply:
Hmm, I was (selfishly) thinking that since we run `tox -e mypy` in a separate environment, it will always use the latest pydantic. But you're right, we also have to think about library users who are running mypy on their own projects (and might have pinned dependencies to pydantic v1 _or_ v2).

I'm uncertain of the best way to handle this, but we probably want to test this branch as a dependency within some live code (that uses mypy) before we merge it. I can try to get to that in a week or so, but if you can sooner, all the better.

Reply:
I used this branch in a project of mine with both `pydantic~=1` and `pydantic~=2`, and the test suite (including mypy) passed in both cases. 🎉